docroot/modules/custom/uiowa_core/uiowa_core.module

```diff
@@ -134,6 +203,168 @@ function template_preprocess_uiowa_core_headline(&$variables) {
 
 }
 
+/**
+ * Preprocess function for card.
+ */
+function template_preprocess_card(&$variables) {
+  $element = $variables['element'];
+
+  // Set up various attributes if not already set up.
+  foreach ([
+    'attributes',
+    'button_attributes',
+    'media_attributes',
+  ] as $attributes_type) {
+    if (!isset($element["#$attributes_type"])) {
+      $element["#$attributes_type"] = [];
+    }
+  }
+
+  // Create a set of title headline classes in case its needed.
+  $headline_classes = ['headline'];
+
+  // Loop through all classes, add any media and headline classes
+  // to the array and remove them from the card classes.
+  if (isset($element['#attributes']['class'])) {
+    foreach ($element['#attributes']['class'] as $key => $style) {
+      if (str_starts_with($style, 'media')) {
+        $element['#media_attributes']['class'][] = $style;
+        unset($element['#attributes']['class'][$key]);
+      }
+      if (str_starts_with($style, 'headline')) {
+        $headline_classes[] = $style;
+        unset($element['#attributes']['class'][$key]);
+      }
+    }
+  }
+
+  $linked_element = FALSE;
+
+  // If there is no URL, then it is not linked.
+  if (!empty($element['#url'])) {
+    if (!is_null($element['#title'])) {
+      $linked_element = 'title';
+    }
+    elseif (isset($element['#link_text'])) {
+      $linked_element = 'button';
+    }
+    elseif (!empty($element['#media'])) {
+      $linked_element = 'media';
+    }
+    else {
+      $linked_element = 'button';
+    }
+  }
+
+  $element['#linked_element'] = $linked_element;
+
+  if (!empty($element['#title'])) {
+    $element['#headline'] = [
+      'headline_text' => $element['#title'],
+      'headline_level' => $element['#title_heading_size'] ?: 'h2',
+      'headline_class' => $headline_classes,
+    ];
+
+    // @todo Set headline_url based on link being set.
```

Comment:
```suggestion
    // @todo Set headline_url based on link being set.
    // https://github.com/uiowa/uiowa/issues/6292
```

Reply:
Based on the code below, I think this is not necessary.